e2e/e2e-script.sh

```diff
@@ -167,7 +164,9 @@ done
 waitForNodeEndTime=$(date +%s)
 log "Waited $((waitForNodeEndTime-waitForNodeStartTime)) seconds for node to join"
 
-# trap 'az vmss delete -g $MC_RESOURCE_GROUP_NAME -n $VMSS_NAME --no-wait' EXIT
+kubectl get nodes
+
+trap 'az vmss delete -g $MC_RESOURCE_GROUP_NAME -n $VMSS_NAME --no-wait' EXIT
```

Comment:
should this step be moved to after the check in L172? i.e should we delete the vmss even if the node failed to joined the cluster before fetching the logs?

Reply:
it's a trap so it only executes when the entire script exits, we fetch logs before deleting the vmss